### ISSUE TITLE
Create Downloader Lambda, To Upload Queue, and Downloaded Data EFS Volume

### ISSUE THREAD

**Issue by ciaransweet** [NASA-IMPACT/hls-sentinel2-downloader-serverless#5]
# TL;DR
The Lambda function that downloads Sentinel 2 Imagery and then populates the upload queue needs to be created

![image](https://user-images.githubusercontent.com/9111975/102629940-c8a80180-4143-11eb-8b5c-a6bff103166c.png)

# Info

Currently the downloader queues up files internally (via an `aria2c` queue) to be downloaded. This is then set up to different hooks which will perform different actions based on their implementation (stop, start, download error, download finish etc). This results in the downloader not downloading files for periods of time (even if there _are_ images available to download).

The new approach to handling downloads will be invoking Lambda from the To Download SQS Queue. Any time there are messages available, Lambda will be invoked.

The IntHub credentials we have are limited to 15 concurrent connections to IntHub for downloads. This is easily enforced by setting the concurrent execution limit on Lambda to 15. At any one time, we will only ever be processing 15 files at once.

To act as an intermediate storage solution (both for use for download/upload and for the ability to easily disconnect sections of the pipeline) we will use an EFS volume to store downloaded imagery until it is uploaded to the correct bucket.

The pseudo code for the flow of the Downloader Lambda is:

```python
# Link Fetcher gets a link and populates the To Download Queue, this then invokes our function:

def handler(event, context):
    download_url, id = get_download_url_and_db_id(event)
    receipt_handle = event['receiptHandle']
    inthub_pass, inthub_username = get_inthub_credentials()
    file_content = requests.get(download_url, auth=(inthub_username, inthub_pass)).content
    write_file_to_efs(file_content)
    update_db(id) # Set the file to downloaded
    confirm_message_received(receipt_handle)
    add_file_to_upload_queue()
```

# Acceptance criteria
* Lambda function and associated methods within are unit tested
* Lambda function is deployed (and limited to 15 concurrent invocations)
* SQS Queue is deployed
* EFS Volume is deployed and attached to the Lambda (Will need to set this up so that we can flag whether to remove the volume on Destroy or keep it and import it in on further deployments)
* Integration test is run which ensures:
    * Function is invoked on the To Download Queue having a message
    * The file is downloaded (Use API Gateway Mock Integrations to return a file)
    * Ensure file is written to EFS (Deploy a lambda in the integration tests and assure this can access the file)
    * Check that `granule` table is updated correctly
    * Check that To Upload Queue has the correct message
